### PR TITLE
Product tests of Kerberized Presto CLI

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java [hunk 1]
```diff
@@ -52,9 +53,37 @@
     private final List<String> nationTableBatchLines;
 
     @Inject
-    @Named("databases.presto.server_address")
+    @Named("databases.presto.cli_server_address")
     private String serverAddress;
 
+    @Inject(optional = true)
```
Comment (losipiuk) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java]: why optional?
Maybe we can force it to be either optional or not depending on list of tests to be executed?

File: presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java [hunk 2]
```diff
@@ -123,14 +152,35 @@ public void shouldRunQueryFromFile()
         temporayFile.deleteOnExit();
         Files.write("select * from hive.default.nation;\n", temporayFile, UTF_8);
 
-        launchPrestoCliWithServerArgument("--file", temporayFile.getAbsolutePath());
+        launchPrestoCliWithServerSpecified("--file", temporayFile.getAbsolutePath());
         assertThat(trimLines(presto.readRemainingOutputLines())).containsAll(nationTableBatchLines);
     }
 
-    private void launchPrestoCliWithServerArgument(String... arguments)
+    private void launchPrestoCliWithServerSpecified(String... arguments)
```
Comment (losipiuk) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java]: why name change?
previous one seems more natural

**Comment by losipiuk** [comment]
So there are no specific tests for kerberos.
We just either test kerberized cli or non-kerberized depending on tempto config?
That is not super cool.

File: presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java (hunk 2 above)
Comment (sopel39) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java]: It seems to be a bit confusing when you look at the method signature if `arguments...` are related to server or not.

**Comment by sopel39** [comment]
We talk about it with @arhimondr. We concluded that it will be consistent with how `tempto` sets up `Hive` JDBC connection. It is either Kerberized on not depending on configuration. Anyway it is not natural to have both unsecured and secured Presto CLI running at the same time.

Additionally, we wanted to avoid hassle of always selecting/excluding kerberized/unsecured tests when running product tests. Now if you have kerberized cluster setup it would just use it with proper configuration

File: presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java (hunk 2 above)
Comment (losipiuk) [inline presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java]: Not convinced. New name seems even more confusing to me.
Whatever. No strong opinion here.

**Comment by losipiuk** [comment]
looks good

**Comment by martint** [comment]
I think this may be covered by https://github.com/prestodb/presto/pull/4867. Can you verify?

If not, can you rebase to resolve the merge conflicts?

**Comment by arhimondr** [comment]
@martint yes, this can be closed.

**Pull request closed without merging**